prototyping/auto-segmentation/sb/05-mini-data-take-2/unet-lung-2d-bn-and-wbce.ipynb

```diff
@@ -0,0 +1,518 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "zi6LRJcDJLRV"
+   },
+   "source": [
+    "# CT scan UNet demo\n",
+    "\n",
+    "This notebook creates a UNet for a minified dataset of animal CTs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "CiSdZ8Q0JLRa"
+   },
+   "outputs": [],
+   "source": [
+    "import pathlib\n",
+    "import urllib.request\n",
+    "import shutil\n",
+    "import collections\n",
+    "import random\n",
+    "\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import tensorflow as tf\n",
+    "import tensorflow.keras.backend as K\n",
+    "import imageio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "9s04q2zZJLRb"
+   },
+   "outputs": [],
+   "source": [
+    "zip_url = 'https://github.com/pymedphys/data/releases/download/mini-lung/mini-lung-medical-decathlon.zip'\n",
+    "zip_filepath = 'data.zip'\n",
+    "\n",
+    "data_directory = pathlib.Path('data', 'unet', 'lung')\n",
+    "\n",
+    "if not data_directory.exists():\n",
+    "    urllib.request.urlretrieve(zip_url, zip_filepath)\n",
+    "    shutil.unpack_archive(zip_filepath, data_directory)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "vUvDaKsUbL1D"
+   },
+   "outputs": [],
+   "source": [
+    "directories = sorted(list(data_directory.glob('*')))\n",
+    "\n",
+    "split = 4\n",
+    "\n",
+    "training_directories = directories[:-split]\n",
+    "validation_directories = directories[-split:]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "QqOS2yMibemH"
+   },
+   "outputs": [],
+   "source": [
+    "assert len(set(validation_directories).intersection(training_directories)) == 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "ydwJnGwmJLRb"
+   },
+   "outputs": [],
+   "source": [
+    "crop_slice = slice(32, -32)\n",
+    "\n",
+    "def _load_image(image_path):\n",
+    "    png_image = imageio.imread(image_path)\n",
+    "    normalised_image = png_image[crop_slice, crop_slice, None] / 255\n",
+    "    \n",
+    "    return normalised_image\n",
+    "\n",
+    "\n",
+    "def _load_mask(mask_path):\n",
+    "    png_mask = imageio.imread(mask_path)\n",
+    "    normalised_mask = png_mask[crop_slice, crop_slice, None] / 255\n",
+    "    \n",
+    "    return normalised_mask"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "FGU8fYdkJLRc"
+   },
+   "outputs": [],
+   "source": [
+    "def load_dataset_type(directories, shuffle=True):\n",
+    "    image_suffix = '_image.png'\n",
```

Comment:
so, this to me is a public function.
And it has no docstring.
But maybe it's not possible to docstring public functions within Jupyter Notebooks.

Reply:
Yeah, it's not very reasonable to be able to import them from another python script. So their not "exported" or "public" in that sense.

There are ways that people do it, by hacking at Python's import logic... but that is not something I am supporting here...